tests/test_report_api.py

```diff
@@ -0,0 +1,175 @@
+
+    norm = hcl.scalar(255/4328)
+
+    return hcl.compute((height, width), lambda x,y: R[x][y] * norm.v, "F")
+
+  s = hcl.create_schedule([imgF, Gx, Gy], sobel_kernel)
+
+  sA = sobel_kernel.A
+  sX = sobel_kernel.X
+  sY = sobel_kernel.Y
+
+  LBX = s.reuse_at(sA._op, s[sX], sX.axis[0], "LBX")
+  LBY = s.reuse_at(sA._op, s[sY], sY.axis[0], "LBY")
+  WBX = s.reuse_at(LBX, s[sX], sX.axis[1], "WBX")
+  WBY = s.reuse_at(LBY, s[sY], sY.axis[1], "WBY")
+  s.partition(LBX, dim=1)
+  s.partition(LBY, dim=1)
+  s.partition(WBX)
+  s.partition(WBY)
+  s.partition(Gx)
+  s.partition(Gy)
+  sP = sobel_kernel.P
+  sR = sobel_kernel.R
+  sF = sobel_kernel.F
+  s[sX].pipeline(sX.axis[1])
+  s[sY].pipeline(sY.axis[1])
+  s[sR].pipeline(sR.axis[1])
+  s[sF].pipeline(sF.axis[1])
+
+  target = hcl.platform.zc706  
+  target.config(compile="vivado_hls", mode="csyn")
+
+  hcl_img = hcl.asarray(img)
+  hcl_Gx = hcl.asarray(np.array([[-1,-2,-1],[0,0,0],[1,2,1]]))
+  hcl_Gy = hcl.asarray(np.array([[-1,0,1],[-2,0,2],[-1,0,1]]))
+  hcl_F = hcl.asarray(np.zeros((height, width)))
+
+  f = hcl.build(s,target) 
+  f(hcl_img, hcl_Gx, hcl_Gy, hcl_F)
+  return f.report()
+
+def parse_rpt():
+  path = pathlib.Path(__file__).parent.absolute()
+  xml_file = str(path) + '/test_report_data/test_csynth.xml'
+  with open(xml_file, "r") as xml:
+    profile = xmltodict.parse(xml.read())["profile"]
+  clock_unit = profile["PerformanceEstimates"]["SummaryOfOverallLatency"]["unit"]
+  summary = profile["PerformanceEstimates"]["SummaryOfLoopLatency"]
+
+  info_table = RptDisp(clock_unit)
+  info_table.scan_range(summary)
+  info_table.get_loops(summary)
+  info_table.init_data(summary)
+  return info_table
+
+def refine(lst):
+  pattern = re.compile(r'\s\s+') 
+  res = []
+  for i in lst:
+    res.append(re.sub(pattern, ', ', i))
+  res[0] = res[0].replace(', ', '', 1)
+  return res
+
+def get_expected(wd):
+  path = pathlib.Path(__file__).parent.absolute()
+  path = str(path) + '/test_report_data/expected.json'
+  with open(path) as f:
+    data = json.loads(f.read())
+  return data[wd]
+
+def test_info(vhls):
+  if vhls:
+    rpt = sobel()
+  else:
+    rpt = parse_rpt()
+  res = rpt.display()
+  res_str = res.split("\n") 
+  lst = refine(res_str)
+  assert lst == get_expected("NoQuery")
+
+def test_loop_query(vhls):
+  if vhls:
+    rpt = sobel()
+  else:
+    rpt = parse_rpt()
+  row_query = ['P', 'A']
+  lq = rpt.display( loops=row_query )
+  lq_str = lq.split("\n") 
+  lq_lst = refine(lq_str)
+  assert lq_lst == get_expected("LoopQuery")
+
+def test_column_query(vhls):
+  if vhls:
+    rpt = sobel()
+  else:
+    rpt = parse_rpt()
+  col_query = ['Latency']
+  cq = rpt.display( cols=col_query )
+  cq_str = cq.split("\n")
+  cq_lst = refine(cq_str)
+  assert cq_lst == get_expected("ColumnQuery")
+
+def test_level_query(vhls):
+  if vhls:  
+    rpt = sobel()
+  else:
+    rpt = parse_rpt()
+  lev_query = 1
```

Comment:
Can we test with different levels?

Reply:
The static data that I am using has `P` with 3 loop nest and other loops have 2 or less. So I thought testing loop level of 1 would best show the querying behavior. I could also change the static dataset to be something different if "1" does not seem to be a good number to use.

Reply:
No. I mean you should test more, in addition to level=1. You should also test other levels.